pytentiostat/config_reader.py

```diff
@@ -0,0 +1,74 @@
+from yaml import load
+try:
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader
+
+def get_output_params():
+    
+    data_out_name = data['general_parameters']['data_output_filename']
+    data_out_path = data['general_parameters']['data_output_path']
+    
+    return data_out_name, data_out_path
+
+def  get_lsv_params():
+    
+    start_voltage = data['linear_sweep_voltammetry']['start_voltage']
+    end_voltage = data['linear_sweep_voltammetry']['end_voltage']
+    sweep_rate = data['linear_sweep_voltammetry']['sweep_rate']
+    
+    return start_voltage, end_voltage, sweep_rate
+
+def get_ca_params():
+    
+    voltage = data['chronoamperometry']['voltage']
+    time = data['chronoamperometry']['time']
+    
+    return voltage, time
+
+def get_cv_params():
+    
+    start_voltage = data['cyclic_voltomoetry']['start_voltage']
+    first_turnover = data['cyclic_voltomoetry']['first_turnover_voltage']
+    second_turnover = data['cyclic_voltomoetry']['second_turnover_voltage']
+    sweep_rate = data['cyclic_voltomoetry']['sweep_rate']
+    cycle_number = data['cyclic_voltomoetry']['number_of_cycles']
+    
+    return start_voltage, first_turnover, second_turnover, sweep_rate, cycle_number
+
+def get_exp_type():
+    
+    exp_type = data['general_parameters']['experiment_type']
+    
+    return exp_type
+
+def get_exp_time():
+    
+    exp_time = data['chronoamperometry']['time']
+    
+    return exp_time
+
+def get_rest_time():
+    
+    rest_time = data['general_parameters']['rest_time']
+    
+    return rest_time
+
+def get_adv_params():
+    
+    conversion_factor = adv_data['conversion_factor']
+    shunt_resistor = adv_data['shunt_resistor']
+    time_step = adv_data['time_step']
+    average_number = adv_data['average_number']
+    time_per_measurement = time_step * 2 * average_number
+    
+    return conversion_factor, shunt_resistor, time_step, average_number, time_per_measurement
+
+#Load the config files
+stream = open('config.yml', 'r')
```

Comment:
Also, for opening files you should use this pattern:
```
with open('config.yml','r') as stream:
    data = load(stream, Loader=Loader)
```
which will automatically close all streams when they are not needed any more.

Reply:
Also, it is recommended to use ``yaml.safe_load``, not ``yaml.load``, the latter being a security vulnerability.  We may not be able to release packages on conda without this change in the future.

Reply:
- I think in the end for config based user-input version, config in same directory as main will be easiest for testing different cases, so this can work for now. Will make variable though after thinking of where would be optimal to put variable config path + whenever trying to implement testing for config.

- New pattern implemented and updated to safe load!